src/sqlitestorage.cpp

```diff
@@ -368,202 +146,50 @@ bool SqliteStorage::load(const QString &uid)
+    QHash<QString, Incidence::List> hash;
+    return d->mBackend.incidences(&hash, uid) && d->addIncidences(hash);
 }
 
 bool SqliteStorage::load(const QDate &start, const QDate &end)
 {
-    if (!d->mDatabase) {
-        return false;
-    }
-
-    // We have no way to know if a recurring incidence
-    // is happening within [start, end[, so load them all.
-    if ((start.isValid() || end.isValid())
-        && !d->loadRecurringIncidences()) {
-        return false;
-    }
-
-    int rv = 0;
-    int count = -1;
-    QDateTime loadStart;
-    QDateTime loadEnd;
-
-    d->mIsLoading = true;
-
+    bool success = true;
+    QDateTime loadStart, loadEnd;
     if (getLoadDates(start, end, &loadStart, &loadEnd)) {
-        const char *query1 = NULL;
-        int qsize1 = 0;
-
-        sqlite3_stmt *stmt1 = NULL;
-        int index = 1;
-        qint64 secsStart;
-        qint64 secsEnd;
-
-        // Incidences to insert
-        if (loadStart.isValid() && loadEnd.isValid()) {
-            query1 = SELECT_COMPONENTS_BY_DATE_BOTH;
-            qsize1 = sizeof(SELECT_COMPONENTS_BY_DATE_BOTH);
-            SL3_prepare_v2(d->mDatabase, query1, qsize1, &stmt1, NULL);
-            secsStart = d->mFormat->toOriginTime(loadStart);
-            secsEnd = d->mFormat->toOriginTime(loadEnd);
-            SL3_bind_int64(stmt1, index, secsEnd);
-            SL3_bind_int64(stmt1, index, secsStart);
-            SL3_bind_int64(stmt1, index, secsStart);
-        } else if (loadStart.isValid()) {
-            query1 = SELECT_COMPONENTS_BY_DATE_START;
-            qsize1 = sizeof(SELECT_COMPONENTS_BY_DATE_START);
-            SL3_prepare_v2(d->mDatabase, query1, qsize1, &stmt1, NULL);
-            secsStart = d->mFormat->toOriginTime(loadStart);
-            SL3_bind_int64(stmt1, index, secsStart);
-            SL3_bind_int64(stmt1, index, secsStart);
-        } else if (loadEnd.isValid()) {
-            query1 = SELECT_COMPONENTS_BY_DATE_END;
-            qsize1 = sizeof(SELECT_COMPONENTS_BY_DATE_END);
-            SL3_prepare_v2(d->mDatabase, query1, qsize1, &stmt1, NULL);
-            secsEnd = d->mFormat->toOriginTime(loadEnd);
-            SL3_bind_int64(stmt1, index, secsEnd);
-        } else {
-            query1 = SELECT_COMPONENTS_ALL;
-            qsize1 = sizeof(SELECT_COMPONENTS_ALL);
-            SL3_prepare_v2(d->mDatabase, query1, qsize1, &stmt1, NULL);
-        }
-        count = d->loadIncidences(stmt1);
+        bool loadAllRecurringIncidences = !isRecurrenceLoaded();
+        QHash<QString, Incidence::List> hash;
+        success = d->mBackend.incidences(&hash, loadStart, loadEnd, loadAllRecurringIncidences) && d->addIncidences(hash);
 
-        if (count >= 0) {
+        if (success) {
             addLoadedRange(loadStart.date(), loadEnd.date());
         }
-        if (loadStart.isNull() && loadEnd.isNull()) {
-            setIsRecurrenceLoaded(count >= 0);
+        if (loadAllRecurringIncidences) {
+           setIsRecurrenceLoaded(success);
```

Comment:
Cosmetics: the indentation got off by one here

Reply:
Oops, corrected.